app/routes/questionnaire.py

```diff
@@ -306,37 +301,12 @@ def _generate_wtf_form(block_schema, schema, current_location):
 
 
 def submit_answers(schema, questionnaire_store, full_routing_path):
-    metadata = questionnaire_store.metadata
-
-    message = json.dumps(
-        convert_answers(schema, questionnaire_store, full_routing_path), for_json=True
-    )
-
-    encrypted_message = encrypt(
-        message, current_app.eq["key_store"], KEY_PURPOSE_SUBMISSION
+    submission_handler = SubmissionHandler(
+        schema, questionnaire_store, full_routing_path
     )
-    sent = current_app.eq["submitter"].send_message(
-        encrypted_message,
-        questionnaire_id=metadata.get("questionnaire_id"),
-        case_id=metadata.get("case_id"),
-        tx_id=metadata.get("tx_id"),
-    )
-
-    if not sent:
-        raise SubmissionFailedException()
-
-    submitted_time = datetime.utcnow()
-    _store_submitted_time_in_session(submitted_time)
+    submission_handler.submit_message()
     get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
-
-    return redirect(url_for("post_submission.get_thank_you"))
-
-
-def _store_submitted_time_in_session(submitted_time):
-    session_store = get_session_store()
-    session_data = session_store.session_data
-    session_data.submitted_time = submitted_time.isoformat()
-    session_store.save()
+    return redirect(submission_handler.get_next_location_url())
```

Comment:
The `get_next_location_url` seems unnecessary given that it's always the same url. It's clearer to see what this is by using `return redirect(url_for("post_submission.get_thank_you"))` here.